### PR TITLE
support redistribute vrf function

### PR DESCRIPTION
**Pull request by zice312963205**
bgpd: Add support for BGP vrf route copying

Example configuration:
    router bgp 100 vrf vrf-1
     address-family ipv4 unicast
      rd vpn export 1:1
	  rt vpn both 1:1
      export vpn
      import vpn
      redistribute vrf vrf-2
     exit-address-family
     !
    !
	router bgp 100 vrf vrf-2
	 neighbor 2.2.2.2 remote-as 1
	 address-family ipv4 unicast
	  rd vpn export 2:2
	  rt vpn both 2:2
      neighbor 2.2.2.2 activate
     exit-address-family
     !
BGP routes learned from the neighbor 2.2.2.2 under VRF-1 can be directly copied to the BGP under VRF-1 without carrying the RD and RT attributes of VRF-1. After being copied to VRF-1, the routes will be sent to VPN with the RD 1:1 and RT 1:1 attributes, forming BGP-VPN routes.

### CODE REVIEW

**Comment by cscarpitta** [comment]
Just a few things we should do before submitting the code:

- Add documentation to the [bgp docs](https://github.com/FRRouting/frr/blob/master/doc/user/bgp.rst) for the new command `redistribute vrf` that you added. This should go in a separate commit.
- Write test cases for the new functionality and add them to the [tests folder](https://github.com/FRRouting/frr/tree/master/tests). This should go in a separate commit.
- Format the code according to the FRR style guidelines. We can use the `clang-format` source formatter tool to automatically format code according to the guidelines.
- Format the commit messages according to the guidelines described [here](https://docs.frrouting.org/projects/dev-guide/en/latest/workflow.html#commit-guidelines), [here](https://docs.frrouting.org/projects/dev-guide/en/latest/workflow.html#commit-messages), and [here](https://docs.frrouting.org/projects/dev-guide/en/latest/workflow.html#signing-off).

I can help with all the above items.

File: bgpd/bgp_vty.c [hunk 1]
```diff
@@ -9363,6 +9364,116 @@ DEFPY(bgp_imexport_vrf, bgp_imexport_vrf_cmd,
...
+		vty_out(vty, "%% Must include route-map name\n");
+		return CMD_WARNING;
+	}
+
+	if (argv_find(argv, argc, "no", &idx))
+		remove = true;
+
+	afi = vpn_policy_getafi(vty, bgp, true);
+	if (afi == AFI_MAX)
+		return CMD_WARNING_CONFIG_FAILED;
```
Comment (cscarpitta) [inline bgpd/bgp_vty.c]: I suggest to replace these lines with something like this:

```C
afi = bgp_node_afi(vty);
safi = bgp_node_safi(vty);
if ((SAFI_UNICAST != safi) || ((AFI_IP != afi) && (AFI_IP6 != afi))) {
	vty_out(vty, "%% redistribute vrf is valid only for unicast ipv4|ipv6\n");
	return CMD_WARNING_CONFIG_FAILED;
}
```
	
and leave the `vpn_policy_getafi` function unchanged.

This should be equivalent to your implementation, but it doesn't change the `vpn_policy_getafi` function which is used in some other places.

**Review by cscarpitta — CHANGES_REQUESTED** [review]

File: bgpd/bgp_vty.c (hunk 1 above)
Comment (zice312963205) [reply bgpd/bgp_vty.c]: Done for this. Thanks for your comments.

File: bgpd/bgp_vty.c [hunk 2]
```diff
@@ -8639,7 +8639,7 @@ static afi_t vpn_policy_getafi(struct vty *vty, struct bgp *bgp, bool v2vimport)
 			"%% context error: valid only in address-family <ipv4|ipv6> unicast block\n");
 		return AFI_MAX;
 	}
-
+#if 0
```
Comment (cscarpitta) [inline bgpd/bgp_vty.c]: Now the function `vpn_policy_getafi()` is never called from your code, right?

I think there is no reason to change it. Can we remove the `#if 0` directive and leave `vpn_policy_getafi()` unchanged?

**Review by cscarpitta — CHANGES_REQUESTED** [review]

File: bgpd/bgp_vty.c (hunk 2 above)
Comment (zice312963205) [reply bgpd/bgp_vty.c]: Ok... I'm sorry, I forgot to restore it.

**Pull request closed without merging**